olive/passes/pytorch/lora.py

```diff
@@ -223,6 +305,12 @@ def _run_for_config(
             )
             config.training_args.evaluation_strategy = "no"
 
+        # Grandient checkpointing seems to be enabled by specifying requires_grad_(True)
+        # for the first parameter of the model.
+        # However, LoRA training does not train that parameter
+        if config.training_args.gradient_checkpointing:
```

Comment:
is this part required for lora in general or only for qlora? For qlora, it is already handles by `prepare_model_for_kbits_training` https://github.com/huggingface/peft/blob/main/src/peft/utils/other.py#L66

Reply:
Yes this is required for lora in general. Qlora handles this but general lora doesn't.

Reply:
I see. I was looking for normal lora fine-tuning examples but couldn't find one that just uses a transformers model.

Reply:
I guess this is specific for llama model maybe? I saw several issue in other github repos discussing this like https://github.com/lm-sys/FastChat/issues/581